### ISSUE TITLE
jsx-indent-props > 'first'option forces you to use 'space' identation

### ISSUE THREAD

**Issue by LunaFidalgo** [yannickcr/eslint-plugin-react#2839]
Hello, 
I'm using eslint with typescript and react. I was trying to use this rule "jsx-indent-props" and I wanted to use the option 'first' but with tab indentation. **The documentation doesn't point that if you select 'first' option you are forced to use 'space' indentation**: 

> It takes an option as the second parameter which can be "tab" for tab-based indentation, a positive number for space indentations or **"first" for aligning the first prop for each line with the tag's first prop**. Note that using the "first" option allows very inconsistent indentation unless you also enable a rule that enforces the position of the first prop.

The code actually is like [that](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/jsx-indent-props.js#L70): 

```js    if (context.options.length) {
      if (context.options[0] === 'first') {
        indentSize = 'first';
        indentType = 'space';
      } else if (context.options[0] === 'tab') {
        indentSize = 1;
        indentType = 'tab';
      } else if (typeof context.options[0] === 'number') {
        indentSize = context.options[0];
        indentType = 'space';
      }
    }
```

Maybe we can add more than one option to reach a combination between 'first' and 'tabs or spaces'? Is there a problem with doing this?

Thank you!

**Comment by ljharb** [comment]
Presumably you're also using indent and jsx-indent, which can easily autofix to your preferred indentation?

**Comment by LunaFidalgo** [comment]
First of all, thank for your response @ljharb! 

> Presumably you're also using indent and jsx-indent, which can easily autofix to your preferred indentation?

Yes, but it's always going to tell me something like this: 

> ESLint: Expected indentation of 6 space characters **but found 0**.(react/jsx-indent-props)

Because I'm not using space for indentation. 

Example: 
![image](https://user-images.githubusercontent.com/36133677/96833987-8a42de80-1441-11eb-9629-b7ed8aa7e0cd.png)

eslintrc.js:
```json
		"indent": [
			"error",
			"tab",
			{
				"ObjectExpression": 1,
				"ignoredNodes":["JSXElement *"]
			}
		],
		"react/jsx-indent-props":["error","first"],
		"react/jsx-indent":["error", "tab"],

`

**Comment by ljharb** [comment]
eslint on the command line makes multiple passes with the autofixer so it'd fix all of those issues at once; i think it's only in editors that make a single pass at a time.
